test/language/optional-chaining/destructuring.js

```diff
@@ -0,0 +1,20 @@
+/*---
+esid: pending
+info: |
+  destructuring is not supported for properties of optional chain
+description: >
+  Static Semantics: IsDestructuring
+    LeftHandSideExpression:
+      CallExpression
```

Comment:
@jridgewell @DanielRosenwasser @rkirsling, I read this rule as stating that destructuring is not supported in an optional chain, I also gathered that this was the outcome of:

https://github.com/tc39/proposal-optional-chaining/issues/74

This test is failing because the current babel implementation supports destructuring.

Am I correct in my reading of this part of the proposal, if so I think we'll want to update the babel implementation.

Reply:
Hmm, it seems that IsDestructuring is only checked from [ForIn/OfBodyEvaluation](https://tc39.es/ecma262/#sec-runtime-semantics-forin-div-ofbodyevaluation-lhs-stmt-iterator-lhskind-labelset), and I think we're just saying that if you see an OptionalExpression as an LHS, then (just like a CallExpression) that's not an example of destructuring.

So I don't know that there's anything that needs to be tested here—note that this test doesn't actually involve an optional chain at present, and the linked idea about optional destructuring is something that would require additional syntax.

Reply:
@rkirsling I understand, it's the concept of "optional destructuring" that's out of scope, and the proposal was something along the lines of `foo?.bar?`; my bad misreading that section of the spec.